vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/resources/META-INF/resources/frontend/comboBoxConnector.js

```diff
@@ -17,6 +17,12 @@ import { ComboBoxPlaceholder } from '@vaadin/combo-box/src/vaadin-combo-box-plac
 
         comboBox.$connector = {};
 
+        // <vaadin-multi-select-combo-box> wraps a regular combo box internally,
+        // to reuse logic between both components we need to identify the
+        // element that implements the data provider mixin
+        const dataProviderMixin =
+          comboBox.localName === 'vaadin-multi-select-combo-box' ? comboBox.$.comboBox : comboBox;
```

Comment:
IIRC, when using `PolymerTemplate` and `@Id`, the `comboBox.$` might be not yet initialized because web component does not yet have shadow root created. See https://github.com/vaadin/vaadin-combo-box-flow/pull/223 

It would be nice to add ITs for `PolymerTemplate` and `LitTemplate` to the `template` folder just to verify that this does not cause bugs, and refactor this logic accordingly if needed.

Reply:
Good catch. This would indeed throw in the case a `ComboBox` is added as a virtual child (as `Dialog` content for example). The shadow root would not get created before the custom element class is finalized.

Reply:
Thanks for catching this, that did indeed not work. Updated the connector to resolve the data provider mixin element on demand. Added ITs for adding the component as a virtual child, through a Lit template, and through a Polymer template.